packages/text-input/src/TextInput.tsx

```diff
@@ -0,0 +1,328 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { uiColors } from '@leafygreen-ui/palette';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { createDataProp } from '@leafygreen-ui/lib';
+
+export const State = {
+  none: 'none',
+  valid: 'valid',
+  error: 'error',
+} as const;
+
+export type State = typeof State[keyof typeof State];
+
+interface TextInputProps {
+  /**
+   * Text shown in bold above the input element.
```

Comment:
nit: when referring to the component in documentation prefer to use capital letters, e.g. Input or <Input />

Reply:
in this particular case, I'm referring to the HTML <input /> element, although I see why that would be unclear. I can still change it, if you want.

Reply:
got it, makes sense